ocaml/typing/printtyped.ml

```diff
@@ -76,11 +76,21 @@ let fmt_constant f x =
   | Const_unboxed_int64 (i) -> fprintf f "Const_unboxed_int64 %Ld" i
   | Const_unboxed_nativeint (i) -> fprintf f "Const_unboxed_nativeint %nd" i
 
-let fmt_mutable_flag f x =
+let fmt_mutable_flag f (x : Asttypes.mutable_flag) =
   match x with
   | Immutable -> fprintf f "Immutable"
   | Mutable -> fprintf f "Mutable"
 
+let fmt_mutable_mode_flag f (x : Types.mutable_flag) =
+  match x with
+  | Immutable -> fprintf f "Immutable"
+  | Mutable m ->
+    if Misc.eq_from_le Mode.Alloc.Const.le m Mode.Alloc.Const.legacy
+    then fprintf f "Mutable"
+    else
+      Misc.fatal_errorf "Unexpected mutable(%a)"
+        Mode.Alloc.Const.print m
```

Comment:
I think you should just print the mode rather than erroring here. This is a debug printer anyway, it's probably better if it can print invalid values.